e2e/tests-dfx/pull.bash

```diff
@@ -68,13 +70,13 @@ teardown() {
 
     ## 1.2. pull onchain canisters in "app" project
     cd ../app
-    assert_command dfx pull dep1
-    assert_match "Pulling canister ryjl3-tyaaa-aaaaa-aaaba-cai...
+    assert_command_fail dfx pull dep1
```

Comment:
Since this is now `assert_command_fail`, will there now be `ERROR:` log output that this test can check for?

Looking at this test now, not knowing a whole lot about what's going on, it's not clear to me why it would be expected to fail.

Reply:
This line now fails because `dfx pull` will fail if any following downloads fails. And in this test, we didn't write `dfx:wasm_url` metadata which definitely will fail.
I would love to add some comments here to clarify the circumstance.

Reply:
Put another way: if `dfx pull` is expected to fail here (which it is), dfx should print some information about the reason.  We should have this test check for that output that describes the failure reason.  Does that make sense?